data/templates/setup_iso.py

```diff
@@ -0,0 +1,150 @@
+import argparse
+import subprocess
+import os
+import errno
+import os.path
+import tempfile
+import atexit
+import sys
+import shutil
+import re
+import urllib
+
+tmpdir = ''
+verbose=1
+
+# Cleanup
+@atexit.register
+def cleanup_tmp():
+    if tmpdir != '':
+        subprocess.check_call(['umount', tmpdir])
+        os.rmdir(tmpdir)
+
+def get_iso_info(fname):
+    # Access ISO to determine OS type from supported list
+    label1=subprocess.Popen(['isoinfo', '-i', fname, '-d'], stdout=subprocess.PIPE)
+    return label1.stdout.read()
+
+def do_install_os(osname, osver, src, dest):
```

Comment:
From your implementation for this function, seems the better name is "setup_repo"